src/frontend/templates/run_stats_head.html

```diff
@@ -0,0 +1,440 @@
+        .domain([xDomainMin, xDomainMax])
+        .range([0, graphSpaceWidth]);
+      xAxis.scale(xScale).ticks(10);
+    };
+    var setXScaleAsDataRows = function() {
+      xScaleIsTime = false;
+      xScaleIsDataRows = true;
+      xScaleIsLogDataRows = false;
+
+      var buffer = 0.01 * (maxDataRows - minDataRows);
+      xDomainMin = minDataRows-buffer;
+      xDomainMax = maxDataRows+buffer*2;
+      xScale =
+        d3.scaleLinear()
+        .domain([xDomainMin, xDomainMax])
+        .range([0, graphSpaceWidth]);
+      xAxis.scale(xScale).ticks(10,"~s");
+    };
+    var setXScaleAsDataRowsLog = function() {
+      xScaleIsTime = false;
+      xScaleIsDataRows = false;
+      xScaleIsLogDataRows = true;
+
+      var maxDataRowsOr10k = Math.max(maxDataRows,10000);
+      var buffer = 0.2 * (maxDataRows - minDataRows);
+      xDomainMin = Math.min(minDataRows + 500000, maxDataRowsOr10k * 0.02);
+      xDomainMax = maxDataRowsOr10k + buffer;
+      xScale =
+        d3.scaleLog()
+        .domain([xDomainMin, xDomainMax])
+        .range([0, graphSpaceWidth]);
+      xAxis.scale(xScale).ticks(15,"~s");
+    };
+
+    setXScaleAsTime();
+
+    var xAxisContent = graphSpace.append("g").attr("transform", "translate(0," + graphSpaceHeight + ")");
+    xAxis(xAxisContent);
+
+    // Takes the xDomain and finds a reasonable y elo range to capture all networks that fall within that range of time.
+    var computeYDomainFromXDomain = function(xDomain) {
+      var filtered;
+      if(xScaleIsTime)
+        filtered = networks.filter(network => network["time"] >= xDomain[0] && network["time"] <= xDomain[1]);
+      else
+        filtered = networks.filter(network => network["total_num_data_rows"] >= xDomain[0] && network["total_num_data_rows"] <= xDomain[1]);
+
+      var minElo = d3.min(filtered.map(network => network["elo"] - Math.min(300, 2.5 * network["elostdev"])));
+      var maxElo = d3.max(filtered.map(network => network["elo"] + Math.min(300, 2.5 * network["elostdev"])));
+      // No elements
+      if(minElo === undefined || maxElo === undefined) {
+        minElo = 0;
+        maxElo = 100;
+      }
+      var eloBuffer = 0.01 * (maxElo - minElo);
+      minElo = minElo - eloBuffer;
+      maxElo = maxElo + eloBuffer;
+      //Also stretch out a little by scale
+      if(maxElo > 0)
+        maxElo *= 1.01;
+      if(minElo > 0)
+        minElo *= 0.99;
+
+      return [minElo, maxElo];
+    };
+
+    // Y axis
+    var yScale =
+        d3.scaleLinear()
+        .domain(computeYDomainFromXDomain(xScale.domain()))
+        .range([graphSpaceHeight, 0]);
+    var yAxis = d3.axisLeft(yScale).tickSize(-graphSpaceWidth*1.0).ticks(20);
+    var yAxisContent = graphSpace.append("g");
+    yAxis(yAxisContent);
+    var unzoomedYDomainMin = yScale.domain()[0];
+    var unzoomedYDomainMax = yScale.domain()[1];
+
+    // Colors
+    var colorScale = d3.scaleSequential(d3.interpolateWarm).domain([0,networkSizes.length-1]);
```

Comment:
Not important: on actual site i find the color not going well with the reste of the site

Maybe we can take bulma colors: https://bulma.io/documentation/overview/colors/